lib/MusicBrainz/Server/Controller/Role/CommonsImage.pm

```diff
@@ -13,16 +13,22 @@ Readonly my $WIKIDATA_PROP_LOCATOR_MAP_IMAGE => "P242";
 after load => sub {
     my ($self, $c) = @_;
 
-    $self->_get_commons_image($c, 1);
+    my $entity = $c->stash->{entity};
+    my $endpoint = $c->uri_for_action($self->action_for('commons_image'), [$entity->gid]);
+    $c->stash->{image} = $self->_get_commons_image($c, 1);
+    $c->stash->{commons_image_props} = $c->json->encode({
+        imageEndpoint => $endpoint->as_string,
+    });
 };
 
 sub commons_image : Chained('load') PathPart('commons-image') {
     my ($self, $c) = @_;
 
-    $self->_get_commons_image($c, 0);
+    my $image = $self->_get_commons_image($c, 0);
 
     $c->res->headers->header('X-Robots-Tag' => 'noindex');
-    $c->stash->{template} = 'components/commons-image.tt';
+    $c->res->content_type('application/json; charset=utf-8');
+    $c->res->{body} = $c->json->encode({image => $image});
```

Comment:
This is the endpoint used to fetch the image info asynchronously. Previously it returned fully rendered HTML. Now it returns JSON to be used as props for the `CommonsImage` component, which is rendered client-side by React in that case.